rainforest/runner.go

```diff
@@ -0,0 +1,77 @@
+package rainforest
+
+import (
+	"errors"
+	"strconv"
+)
+
+// RunParams is a struct holding all potential parameters needed to start a new RF run.
+type RunParams struct {
+	Tests         []int    `json:"tests,omitempty"`
+	Tags          []string `json:"tags,omitempty"`
+	SmartFolderID int      `json:"smart_folder_id,omitempty"`
+	SiteID        int      `json:"site_id,omitempty"`
+	Crowd         string   `json:"crowd,omitempty"`
+	Conflict      string   `json:"conflict,omitempty"`
+	Browsers      []string `json:"browsers,omitempty"`
+	Description   string   `json:"description,omitempty"`
+	EnvironmentID int      `json:"environment_id,omitempty"`
+}
+
+// RunStatus represents a status of a RF run in progress.
+type RunStatus struct {
+	ID           int    `json:"id"`
+	State        string `json:"state"`
+	StateDetails struct {
+		Name         string `json:"name"`
+		IsFinalState bool   `json:"is_final_state"`
+	} `json:"state_details"`
+	Result          string `json:"result"`
+	CurrentProgress struct {
+		Percent  int `json:"percent"`
+		Total    int `json:"total"`
+		Complete int `json:"complete"`
+		NoResult int `json:"no_result"`
+	} `json:"current_progress"`
+	FrontendURL string `json:"frontend_url,omitempty"`
+}
+
+// CreateRun starts a new RF run with given params.
+func (c *Client) CreateRun(params RunParams) (RunStatus, error) {
+	var runStatus RunStatus
+
+	// Usual stuff - create a request and send it
+	req, err := c.NewRequest("POST", "runs", params)
+	if err != nil {
+		return runStatus, err
+	}
+	_, err = c.Do(req, &runStatus)
+	if err != nil {
+		return runStatus, err
+	}
+
+	// Here we check if we successfully created a run in RF, if not then bail with an error.
+	if runStatus.ID == 0 {
+		return runStatus, errors.New("Failed to create a run.")
```

Comment:
prolly show the error. 

also, correct way to check is status code, not this way